src/commands/referral/referralRandom.ts

```diff
@@ -0,0 +1,51 @@
+import { SlashCommandSubcommandBuilder } from '@discordjs/builders';
+import { getPrismaClient } from '../../clients';
+import { getRandomIntInclusive } from '../../utils';
+import { AutocompleteHandler, CommandHandler } from '../command';
+import { cleanupExipredCode } from './cleanupExipredCode';
+import { searchServices } from './services';
+
+export const data = new SlashCommandSubcommandBuilder()
+  .setName('random')
+  .setDescription('randomly get a code/link for a service')
+  .addStringOption((option) =>
+    option
+      .setName('service')
+      .setDescription(
+        'service to refer(type more than 3 characters to see suggestion)'
+      )
+      .setRequired(true)
+      .setAutocomplete(true)
+  );
+
+export const autocomplete: AutocompleteHandler = async (interaction) => {
+  const searchTerm = interaction.options.getString('service', true);
+  if (searchTerm.length < 4) return;
+
+  const options = searchServices(searchTerm);
+  interaction.respond(options);
+};
+
+export const execute: CommandHandler = async (interaction) => {
+  const service =
+    interaction.options.getString('service', true)?.trim().toLowerCase() ?? '';
+
+  const prisma = getPrismaClient();
+  const referrals = await prisma.referralCode.findMany({
+    where: {
+      service: {
+        contains: service,
+      },
+    },
+  });
+
+  const filteredReferrals = await cleanupExipredCode(referrals);
```

Comment:
anh nghĩ cái này k cần await, clean up expired stuff thì cứ để nó running in the background được rồi :-/ Hay ý em là make sure nó đc clean trước khi random?